cpp/src/arrow/util/span_test.cc

```diff
@@ -51,7 +51,6 @@ TEST(Span, Construction) {
 
   static_assert(std::is_constructible_v<span<const int>, decltype(arr)&>);
   static_assert(std::is_constructible_v<span<const int>, decltype(const_arr)&>);
-  static_assert(std::is_constructible_v<span<const int>, span<int>>);
```

Comment:
Hmm... does this mean conversion from `span<T>` to `span<const T>` doesn't work anymore?

Reply:
Nope. Sorry I meant to leave this comment on this PR but forgot, but the reason I removed this line is because it's redundant. The same check already exists on line 42. I noticed it while experimenting.

Reply:
Ok, thanks!